src/plugins/Network.js

```diff
@@ -47,26 +60,69 @@ module.exports = class Network extends OverlayPlugin {
-            if (values.length < 9) {
+        this.instance.registerEventCallback('network_profile', this.handleNetworkProfile.bind(this));
+    }
+
+    // Handle settings event to enable/disable wifi|mobile data
+    handleSettings(message) {
+        const values = message.split(' ');
+
+        if (values[0] === 'if') {
+            this.wifiInput.disabled = false;
+            this.mobileInput.disabled = false;
+
+            if (values.length !== 3) {
                 return;
             }
-            const upSpeed = values[2].split(':');
-            const downSpeed = values[3].split(':');
-            const upDelay = values[4].split(':');
-            const downDelay = values[5].split(':');
-            const upPacketLoss = values[6].split(':');
-            const downPacketLoss = values[7].split(':');
-            const dnsDelay = values[8].split(':');
-
-            const isThrottlingEnabled = upSpeed[1] === 'enabled'
-                && downSpeed[1] === 'enabled'
-                && upDelay[1] === 'enabled'
-                && downDelay[1] === 'enabled'
-                && upPacketLoss[1] === 'enabled'
-                && downPacketLoss[1] === 'enabled'
-                && dnsDelay[1] === 'enabled';
+
+            const wifiOn = values[1].match(/(wifi:)(\w+)/);
+            if (wifiOn) {
+                this.wifiInputChecked = wifiOn[2] === 'on';
+            }
+            const mobileOn = values[2].match(/(mobile:)(\w+)/);
+            if (mobileOn) {
+                this.mobileInputChecked = mobileOn[2] === 'on';
+            }
+
+            if (this.widgetRendered) {
+                this.wifiInput.checked = this.wifiInputChecked;
+                this.mobileInput.checked = this.mobileInputChecked;
+                this.updateMobileSectionStatus();
+            }
+        } else if (values[0] === 'parameter' && values.length > 2 && values[2].includes('android_version')) {
+            const version = values[2].match(/(android_version:)(\w+)/);
+            this.androidVersion = version[2];
+            if (!this.widgetRendered) {
+                this.renderWidget();
+            }
+        }
+    }
+
+    // Update network details (downSpeed, downDelay...) according to the selected network type.
+    handleNetworkProfile(message) {
+        const values = message.split(' ');
+        if (this.androidVersion < ANDROID_8 && (values.length < 9 || values[1] === 'phone')) {
+            return;
+        } else if (this.androidVersion >= ANDROID_8 && (values.length < 11 || values[1] === 'wifi')) {
+            return;
+        }
+        const upSpeed = values[2].split(':');
+        const downSpeed = values[3].split(':');
+        const upDelay = values[4].split(':');
+        const downDelay = values[5].split(':');
+        const upPacketLoss = values[6].split(':');
+        const downPacketLoss = values[7].split(':');
+        const dnsDelay = values[8].split(':');
+
+        if (this.androidVersion < ANDROID_8) {
```

Comment:
Please use the capability we created for that instead of hardcoding this in the UI https://github.com/Genymobile/genymotion_private_device_genymotion/blob/genymotion-8.0.0_r11/vbox86/device.mk#L191